app/models/transformation_mapping/vm_migration_validator.rb

```diff
@@ -0,0 +1,144 @@
+class TransformationMapping::VmMigrationValidator
+  require 'miq-hash_struct'
+
+  VM_CONFLICT = "conflict".freeze
+  VM_EMPTY_NAME = "empty_name".freeze
+  VM_IN_OTHER_PLAN = "in_other_plan".freeze
+  VM_INACTIVE = "inactive".freeze
+  VM_INVALID = "invalid".freeze
+  VM_MIGRATED = "migrated".freeze
+  VM_NOT_EXIST = "not_exist".freeze
+  VM_VALID = "ok".freeze
+
+  def initialize(mapping, vm_list = nil)
+    @mapping = mapping
+    @vm_list = vm_list
+  end
+
+  def validate
+    @vm_list.present? ? identify_vms : select_vms
+  end
+
+  def select_vms
+    valid_list = []
+
+    Vm.where(:ems_cluster => mapped_clusters).includes(:lans, :storages).each do |vm|
+      reason = validate_vm(vm, true)
+      valid_list << VmMigrateStruct.new(vm.name, vm, VM_VALID, reason) if reason == VM_VALID
+    end
+
+    {"valid" => valid_list}
+  end
+
+  def identify_vms
```

Comment:
@kbrock @bdunne @Fryguy @NickLaMuro Could you please review this method? Thanks.